src/azure-cli-core/azure/cli/core/profiles/__init__.py

```diff
@@ -89,3 +89,15 @@ def get_sdk(cli_ctx, resource_type, *attr_args, **kwargs):
     'latest': AZURE_API_PROFILES['latest'],
     '2017-03-09-profile': AZURE_API_PROFILES['2017-03-09-profile']
 }
+
+
+def augment_api_profiles(profile_name, resource_type, api_version):
```

Comment:
So if I wanted to add a resource type to multiple profiles I would have:
```
augment_api_profiles('latest', MY_RESOURCE_TYPE, '2016-05-01')
augment_api_profiles('2017-01-01-profile', MY_RESOURCE_TYPE, '2016-05-01')
...
```

A more concise syntax to consider:
```
register_resource_type(MY_RESOURCE_TYPE, {
  'latest': '2016-05-01',
  '2017-01-01-profile': '2016-05-01'
})
```
Essentially you specify the resource type once and then a dictionary of `profile: version` pairs.

Reply:
Requiring a dictionary argument seems confusing.

Reply:
I agree with @williexu, the dict format seems confusing and that was an original comment in my first implementation that allowed users to pass in a dict.

I could potentially change the method name from `augment_api_profiles` to `register_resource_type` if that's better for everyone.

Reply:
yes to `register_resource_type`

Reply:
I didn't see anything wrong with the dictionary implementation in the original. I don't like that I would have to repeat so much code repeatedly calling "augment_api_profiles" (or "register_resource_type") once for each resource type and each profile.  I'd be curious what @troydai thinks here. 

Something that looks closest to the API profile structure seems most logical to me (I have a feeling that's probably what your original implementation looked like--but I didn't have problems with that).

```Python
register_resource_types({
  'latest': {
   MY_RT_1: '2018-01-01',
   MY_RT_2: '2018-01-01'
  }
  '2018-01-01-profile': {
   MY_RT_1: '2018-01-01',
   MY_RT_2: '2018-01-01'
  }
})
```

It's essentially just a profile dictionary merge. I don't think that's necessarily confusing--just would just need to validate it.

Reply:
@tjprescott  why not:
```
for  profile in 'latest', '2018-...':
    for rt in RT_1, RT_2:
          register_resource_type(profile, rt, version)
```